go/private/library.bzl

```diff
@@ -113,63 +114,52 @@ def emit_library_actions(ctx, srcs, deps, cgo_object, library, want_coverage):
   transformed = dict_of(source)
   transformed["go"] = go_srcs
 
-  return struct(
-    label = ctx.label,
-    files = depset([out_lib]),
-    library = out_lib,
-    race = race_lib,
-    searchpath = searchpath,
-    searchpath_race = searchpath_race,
-    runfiles = runfiles,
-    srcs = join_srcs(struct(**transformed)),
-    importpath = importpath,
-    cgo_object = cgo_object,
-    direct_deps = deps,
-    transitive_cgo_deps = transitive_cgo_deps,
-    transitive_go_libraries = transitive_go_library_deps + [out_lib],
-    transitive_go_libraries_race = transitive_go_library_deps_race + [race_lib],
-    transitive_go_library_paths = transitive_go_library_paths,
-    transitive_go_library_paths_race = transitive_go_library_paths_race,
-    gc_goopts = gc_goopts,
-  )
+  return [
+    GoLibrary(
+          library = out_lib,
+          race = race_lib,
+          searchpath = searchpath,
+          searchpath_race = searchpath_race,
+          srcs = join_srcs(struct(**transformed)),
+          importpath = importpath,
+          direct_deps = deps,
+          transitive_cgo_deps = transitive_cgo_deps,
+          transitive_go_libraries = transitive_go_library_deps + [out_lib],
+          transitive_go_libraries_race = transitive_go_library_deps_race + [race_lib],
+          transitive_go_library_paths = transitive_go_library_paths,
+          transitive_go_library_paths_race = transitive_go_library_paths_race,
+          gc_goopts = gc_goopts,
+          runfiles = runfiles,
+      ),
+      CgoLibrary(
+          object = cgo_object,
+      ),
+  ]
 
 def _go_library_impl(ctx):
   """Implements the go_library() rule."""
   cgo_object = None
   if hasattr(ctx.attr, "cgo_object"):
     cgo_object = ctx.attr.cgo_object
-  lib_result = emit_library_actions(ctx,
+  golib, cgolib = emit_library_actions(ctx,
       srcs = ctx.files.srcs,
       deps = ctx.attr.deps,
       cgo_object = cgo_object,
       library = ctx.attr.library,
       want_coverage = ctx.coverage_instrumented(),
   )
 
+  depset([golib])
```

Comment:
Nope, was left over from me trying to work out which fields were mutable as early as possible.
Thanks!